src/gerbil/gxc-main.ss

```diff
@@ -9,6 +9,7 @@
   (displayln " -h,--help                   display this help message and exit")
   (displayln " -d <dir>                    set compiler output directory; defaults to $GERBIL_PATH/lib")
   (displayln " -exe                        compile an executable")
+  (displayln " -target                     the compilation output target: C or js; defaults to C")
```

Comment:
I wouldnt limit to js or C, it can be anything the user has enabled in gambit.

So drop the "C or js", just say defaults to C.

Reply:
There is no "gambit", so unless a user went into our config, added the target, compiled the base system and modules for that language by modifying our build.sh, and ported our runtime over to that language so that gxc supports either making a module or exectuable there is only C or JS supported by gxc. 

We are not gambit. The users often do not know we build off of gambit. Having totally unsupported things in our `--help` is kind of off base. 

And, if the user knows they can ask gambit for anything they can trivially pass `-gsc-option -target python` even though it is not supported by Gerbil. 

So I think it's better for the --help to say what we do support, not what gambit may or may not support that will most likely not at all work with gerbil. 

So, respectfully disagree :)

Reply:
eait, we can configure different targets during configure! Thats the gambit i refer to.

Reply:
So I need to add the ability to configure different targets to our configure? Then say in the gxc help that you can output to any target even though our expander and runtime only work in C or JS?

I can do that, no problem, but I think that confusing users and forcing them to see what does not work is not part of the `--help` idea. 

Here's the plan? 

1) `readonly gerbil_targets=""` = no longer read only and a new `--enable-targets` conf options? 
2) Change the help to "Any target specifically enabled and/or C"
3) Add "Will not work without porting" and/or "Try to modify the source code and add a compilation-target? cond expand for the target you need to work and rebuild" along with "try to port only the bootstrap first as the current run time, interpreter and compiler will never work with your new target without changes and the bootstrap is the easy way to ensure"
4) Also document "JavaScript is the only working target that is supported if that target is passed, the rest are 'here by dragons' both for Gerbil and the Gambit Universtal Backend" somewhere? 

That still seems pointless and confusing to me :) Am a totally looking at this from the wrong point of view?

Reply:
We dont have to do it for this pr.

Having said that, we should add explicit support for altwtnative targets we support and tailor the build process, say `--enable-js` and do thr js build after the C build, possibly placing artifacts in an alternative directory.

The main thing is that i dont want to set our choices in stone, so C we support, js we will soon support and who knows what else we will support in the future? Quite possibly the native backend when it is mature enough. And why discourage experiments?

So lets keep it simple, "defaults to C" without constraining other options.

Reply:
again, not in this pr. Just remove the "js or c" for now.

Reply:
> Having said that, we should add explicit support for alternative targets we support and tailor the build process, say `--enable-js` and do the js build after the C build, possibly placing artifacts in an alternative directory.

That's the plan, same page. 

> The main thing is that i dont want to set our choices in stone, so C we support, js we will soon support and who knows what else we will support in the future? Quite possibly the native backend when it is mature enough. And why discourage experiments?

Not discouraging experiments in the least! But also trying to show that, after this PR, we fully support JS, it's not and experiment, no confusion, and is not a "this does not work but go ahead and try". 

> So lets keep it simple, "defaults to C" without constraining other options.

How about "C and js are fully supported, can be any other backend; defaults to C" ? 

That way we allow confusion and experimentation, no constraints, but specify that JS is fully supported to ease any confusion a user may have about it.

No confusion, no constraints, and mentioning a major feature that most other scheme implementations are nowhere near supporting. 

Make sense?

Reply:
Too many words? ;)

But maybe its fine. We can also have it in the documentation.